libsolidity/ast/Types.h

```diff
@@ -1187,6 +1203,7 @@ class UserDefinedValueType: public Type
 
 	std::string toString(bool _withoutDataLocation) const override;
 	std::string canonicalName() const override;
+	std::string eip712TypeName() const override { solAssert(false, "EIP-712 is not supported for struct members of user defined value type"); }
```

Comment:
Why not? I think it makes sense to allow this, and keep it similar to how contract types are handled, i.e., use the underlying type.

Here are some discussions on this: https://twitter.com/_prestwich/status/1686501345732460554

Reply:
Should we consider enums as uint256-wrapping structs? Same question about UDT...

Reply:
Fixed this